tools/ci-build/changelogger/src/render.rs

```diff
@@ -494,18 +507,189 @@ fn render(
+
+**Contributors**
+Thank you for your contributions! ❤
+- @another-contrib ([smithy-rs#200](https://github.com/smithy-lang/smithy-rs/issues/200))
+- @external-contrib ([smithy-rs#446](https://github.com/smithy-lang/smithy-rs/issues/446), [smithy-rs#447](https://github.com/smithy-lang/smithy-rs/issues/447))
+- @other-external-dev ([smithy-rs#446](https://github.com/smithy-lang/smithy-rs/issues/446), [smithy-rs#447](https://github.com/smithy-lang/smithy-rs/issues/447))
+
+"#;
+
+    const AWS_SDK_EXPECTED_END_TO_END: &str = r#"v0.1.0 (January 4th, 2022)
+==========================
+**Breaking Changes:**
+- :warning: ([smithy-rs#445](https://github.com/smithy-lang/smithy-rs/issues/445)) I made a major change to update the AWS SDK
+
+**New this release:**
+- :tada: ([smithy-rs#446](https://github.com/smithy-lang/smithy-rs/issues/446), @external-contrib) I made a change to update the code generator
+
+**Service Features:**
+- `aws-sdk-ec2` (0.12.0): Some API change
+- `aws-sdk-s3` (0.14.0): Some new API to do X
+
+**Service Documentation:**
+- `aws-sdk-ec2` (0.12.0): Updated some docs
+
+**Contributors**
+Thank you for your contributions! ❤
+- @external-contrib ([smithy-rs#446](https://github.com/smithy-lang/smithy-rs/issues/446))
+
+"#;
+
+    #[test]
+    fn end_to_end_changelog_entry_markdown_files() {
+        let temp_dir = TempDir::new().unwrap();
+        let smithy_rs_entry1 = r#"---
+applies_to: ["client", "server"]
+authors: ["rcoh", "jdisanti"]
+references: ["smithy-rs#445"]
+breaking: true
+new_feature: false
+bug_fix: false
+---
+I made a major change to update the code generator
+"#;
+        let smithy_rs_entry2 = r#"---
+applies_to: ["client", "server"]
+authors: ["external-contrib", "other-external-dev"]
+references: ["smithy-rs#446", "aws-sdk#123"]
+breaking: false
+new_feature: true
+bug_fix: false
+---
+I made a change to update the code generator
+"#;
+        let smithy_rs_entry3 = r#"---
+applies_to: ["client", "server"]
+authors: ["another-contrib"]
+references:  ["smithy-rs#200"]
+breaking: false
+new_feature: false
+bug_fix: false
+---
+I made a minor change
+"#;
+        let aws_sdk_entry1 = r#"---
+applies_to: ["aws-sdk-rust"]
+authors: ["rcoh"]
+references: ["smithy-rs#445"]
+breaking: true
+new_feature: false
+bug_fix: false
+---
+I made a major change to update the AWS SDK
+"#;
+        let aws_sdk_entry2 = r#"---
+applies_to: ["aws-sdk-rust"]
+authors: ["external-contrib"]
+references: ["smithy-rs#446"]
+breaking: false
+new_feature: true
+bug_fix: false
+---
+I made a change to update the code generator
+"#;
+        let smithy_rs_entry4 = r#"---
+applies_to: ["client", "server"]
+authors: ["external-contrib", "other-external-dev"]
+references: ["smithy-rs#446", "smithy-rs#447"]
+breaking: false
+new_feature: true
+bug_fix: false
+---
+I made a change to update the code generator
+
+**Update guide:**
+blah blah
+"#;
+
+        // We won't handwrite changelog entries for model updates, and they are still provided in
```

Comment:
Great question, in fact, we can switch to the JSON format for model updates (seems like we didn't have the JSON format when the code was written to support model updates in catapult in early days, which was why TOML was chosen at the time). Switching to JSON can simplify things because
- `changelogger` can remove the use of the `toml` crate
- our internal code doesn't need to use a TOML parsing library anymore

Sounds like there will be the 3rd PR for dropping the `TOML` format from `changelogger` 👍